### DISCUSSION TITLE
Import Connections from .csv

### DISCUSSION THREAD

**Discussion by FlogDonkey** [dbeaver/dbeaver#21738] [Q&A]
Documentation states connections can be imported via .csv or XML.
I have a bunch of connections I've saved in .csv, as per the documentation specification, but documentation does not specify (and I have been unable to find) any way to import these connections into DBeaver. Is this option not enabled by default? How does one go about actually importing these connections?

**Comment by hieuxlu** [comment]
I got the same problem and found out that even though this is not mentioned in the Wiki, `data-sources.json` actually supports specifying plain text `user, password` attributes in connection property. This means we can write a script to convert any format to data-sources.json with `user, password `attributes, to import in projects. 

I think for obvious security reasons, this was not mentioned in the Wiki, but it’s still very handy in most scenarios where we do not need encrypted credentials.

**Comment by fbgeobit** [comment]
Hi All:
You can import connections stored in csv files following this documentation: 
1. Create a csv file with connection properties: https://dbeaver.com/docs/dbeaver/Admin-Manage-Connections/
2. Import it on DBeaver: https://github.com/dbeaver/dbeaver/wiki/How-to-import-Connections-from-External-Tools#custom-connections-format

It's really difficult to find.
